### PR TITLE
More traits fixes

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -393,6 +395,10 @@ mod validation_logic {
 						return false
 					}
 
+					if trait_struct.records.len() == 0 {
```
Comment (aang114) [inline runtime/src/lib.rs:398]: good catch

File: pallets/protos/src/lib.rs [hunk 2]
```diff
@@ -170,7 +170,7 @@ pub mod pallet {
 
 	/// **StorageMap** that maps a **Trait ID** to the **name of the Trait**
 	#[pallet::storage]
-	pub type Traits<T: Config> = StorageMap<_, Identity, Hash64, Vec<u8>, ValueQuery>;
+	pub type Traits<T: Config> = StorageMap<_, Identity, Hash64, Vec<u8>, OptionQuery>;
```
Comment (aang114) [inline pallets/protos/src/lib.rs:173]: what does `OptionQuery` do as opposed to `ValueQuery` btw?

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -414,45 +420,56 @@ mod validation_logic {
...
-									shards_info.implementing.contains(shards_trait),
+									shards_info.implementing.contains(trait_hash),
 								_ => false,
 							}
 						} else {
 							false
 						}
-					})
+					}) {
+						if let Some(_) = pallet_protos::Traits::<Runtime>::get(trait_hash) {
```
Comment (aang114) [inline runtime/src/lib.rs]: isn't this check redundant?

If one of the references has the `trait_hash` in its "implementing" field, obviously it exists, no?

**Review by aang114 — CHANGES_REQUESTED** [review]
There is one redundant check

File: pallets/protos/src/lib.rs (hunk 2 above)
Comment (sinkingsugar) [reply pallets/protos/src/lib.rs:173]: `get` returns `Option<_>` instead of default

File: runtime/src/lib.rs (hunk 3 above)
Comment (sinkingsugar) [reply runtime/src/lib.rs]: true

File: scripts/prometheus.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+global:
```
Comment (aang114) [inline scripts/prometheus.yaml:1]: what's prometheus.yaml for?

**Review by aang114 — APPROVED** [review]
LGTM

File: scripts/prometheus.yaml (hunk 4 above)
Comment (sinkingsugar) [reply scripts/prometheus.yaml:1]: to run prometheus

**Pull request merged**